### PR TITLE
Lots of changes

### PR DESCRIPTION
**Pull request by ethanrcampbell02**
Here's a list of all the major things:
- scripts can be run from any directory
- added utils folder for minor standalone scripts
- print more information about undefined bits (potential tiles and resources in tiles)
- added some license boiler plates that were missing
- added correct handling for edge case CLB bits
- revised affected resource tracing for Vivado interface to correctly
- xc7a35t support

### CODE REVIEW

File: lib/statistics.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def get_bit_group_stats(group_bits:dict, print_flag = False, outfile: TextIOWrap
         tile, _, _, _, change, fault_error, _, _ = group_bits[fault_bit]
 
         # Update the statistics based on the current fault bit's tile
-        if tile == 'NA':
+        if 'NA' in tile:
```
Comment (JacksonSahleen) [inline lib/statistics.py:146]: I'm not sure why this change would need to be made. It seems to me like it would be more correct to check if the tile is NA instead of just if it contains NA to avoid random misinterpretations

File: lib/design_query.py [hunk 2]
```diff
@@ -28,6 +28,7 @@
 from subprocess import Popen, PIPE, STDOUT
 from threading import Thread
 from queue import Queue, Empty
+import re
```
Comment (JacksonSahleen) [inline lib/design_query.py:31]: This is an unused import probably added automatically by VSCode. Should be removed though.

**Review by JacksonSahleen — CHANGES_REQUESTED** [review]
I have a question about the handling of NA returns in statistics.py, and there is an unused import in design_query.py that should probably be removed before merging.

File: lib/statistics.py (hunk 1 above)
Comment (ethanrcampbell02) [reply lib/statistics.py:146]: I believe I made that change in a previous iteration of the changes to undefined bits where the `tile` field for an undefined bit would be an array instead of just a simple string. So that is why that it is the way it is, though I will still fix that.

**Review by JacksonSahleen — APPROVED** [review]
